src/Client/Web/Bit.Client.Web.BlazorUI/Components/NumericTextField/BitNumericTextField.razor.cs

```diff
@@ -432,13 +432,13 @@ private async Task HandleFocus(FocusEventArgs e)
             }
         }
 
-        private int CalculatePrecision(double value)
+        private int CalculatePrecision(TValue? value)
         {
             var pattern = isDecimals ? @"[1-9]([0]+$)|\.([0-9]*)" : @"(^-\d+$)|\d+";
             var regex = new Regex(pattern);
-            if (regex.IsMatch(value.ToString(CultureInfo.InvariantCulture)) is false) return 0;
+            if (regex.IsMatch($"{value}") is false) return 0;
```

Comment:
what is this change for?
I think the previous version was much better.

Reply:
because the value is a generic type and is not support ToString wite `CultureInfo.InvariantCulture` argument.

Reply:
oh right,
the next version of C# will resolve these types of issues with static abstract members for interfaces.